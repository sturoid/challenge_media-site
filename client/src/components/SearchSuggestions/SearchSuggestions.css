:local(.responsive-table) {
  overflow-x: auto;
}

:local(table) {
  margin: 0 auto;
  padding: 0;
  width: 100%;
}

:local(thead) {
  border-bottom: 1px solid #e6e7e9;
}

:local(th) {
  color: #777;
  padding: 10px;
}

:local(tbody tr) {
  border-bottom: 1px solid #e6e7e8;
}

:local(tbody tr td) {
  text-align: center;
  padding: 10px 0;
}

@media (min-width: 768px) {
  :local(tbody tr td) {
    text-align: left;
  }
}
