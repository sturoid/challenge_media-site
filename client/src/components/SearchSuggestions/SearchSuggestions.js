import React from 'react';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { tableResponsive, table } from './SearchSuggestions.styles';
import searchDataType from '../../lib/propTypes';

class SearchSuggestions extends React.Component {
  state = {
    toRoute: null
  };

  handleClick = (e, value) => {
    this.setState({
      toRoute: value
    });
  };

  render() {
    const { results, fields } = this.props;

    if (this.state.toRoute) {
      return <Redirect to={`/user/${this.state.toRoute}`} />;
    }

    return (
      <React.Fragment>
        {results &&
          results.length && (
            <div className={tableResponsive}>
              <table className={table}>
                <thead>
                  <tr>
                    {fields.map(field => (
                      <th key={field.name}>{field.name}</th>
                    ))}
                  </tr>
                </thead>
                <tbody>
                  {results.map(result => (
                    <tr
                      key={result.id}
                      onClick={e => this.handleClick(e, result.id)}
                    >
                      {fields.map(field => (
                        <td key={field.name}>{result[field.value]}</td>
                      ))}
                    </tr>
                  ))}
                </tbody>
              </table>
            </div>
          )}
      </React.Fragment>
    );
  }
}

SearchSuggestions.propTypes = {
  results: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired,
  fields: PropTypes.arrayOf(searchDataType).isRequired
};

export default SearchSuggestions;
