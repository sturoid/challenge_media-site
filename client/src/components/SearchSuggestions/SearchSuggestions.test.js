import React from 'react';
import renderer from 'react-test-renderer';
import SearchSuggestions from './SearchSuggestions';

const searchResults = [
  {
    id: 1,
    name: 'Stuart Clove',
    age: 33,
    state: 'Utah',
    dontShow: 'Should not show'
  },
  {
    id: 2,
    name: 'Jo Whitney',
    age: 27,
    state: 'Utah',
    dontShow: 'Should not show'
  },
  {
    id: 3,
    name: 'Cooper Whitney',
    age: 7,
    state: 'Florida',
    dontShow: 'Should not show'
  }
];

const showFields = [
  { name: 'Name', value: 'name' },
  { name: 'Age', value: 'age' },
  { name: 'State', value: 'state' }
];

describe('<SearchSuggestions />', () => {
  it('renders correctly and equals snapshot', () => {
    const tree = renderer
      .create(<SearchSuggestions results={searchResults} fields={showFields} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
