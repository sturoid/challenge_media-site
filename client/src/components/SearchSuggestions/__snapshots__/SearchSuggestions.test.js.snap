// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchSuggestions /> renders correctly and equals snapshot 1`] = `
<div
  className="css-qijqp5"
>
  <table
    className="css-1z0wrym"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Location
        </th>
        <th>
          Occupation
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        onClick={[Function]}
      >
        <td>
          Stuart Clove
        </td>
        <td />
        <td />
      </tr>
      <tr
        onClick={[Function]}
      >
        <td>
          Jo Whitney
        </td>
        <td />
        <td />
      </tr>
      <tr
        onClick={[Function]}
      >
        <td>
          Cooper Whitney
        </td>
        <td />
        <td />
      </tr>
    </tbody>
  </table>
</div>
`;
